tests/vmi_kernel_boot_test.go

```diff
@@ -122,4 +124,65 @@ var _ = Describe("[sig-compute]VMI with external kernel boot", func() {
 			Expect(err.Error()).To(ContainSubstring("denied the request: spec.domain.firmware.kernelBoot.container must be defined with at least one of the following: kernelPath, initrdPath"))
 		})
 	})
+
+	Context("with external alpine-based kernel only (without initrd)", func() {
+		getVMIKernelBoot := func() *kubevirtv1.VirtualMachineInstance {
+			vmi := utils.GetVMIKernelBoot()
+			// Remove initrd path from vmi spec
+			kernelBoot := vmi.Spec.Domain.Firmware.KernelBoot
+			kernelBoot.Container.InitrdPath = ""
+			return vmi
+		}
+
+		It("ensure successful boot", func() {
+			vmi := getVMIKernelBoot()
+
+			obj, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
+			Expect(err).ToNot(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(obj)
+		})
+
+		It("ensure successful boot and deletion when VMI has a disk defined", func() {
+			By("Creating VMI with disk and kernel boot")
+			vmi := libvmi.NewAlpine(libvmi.WithResourceMemory("1Gi"))
+
+			utils.AddKernelBootToVMI(vmi)
+			// Remove initrd path from vmi spec
+			kernelBoot := vmi.Spec.Domain.Firmware.KernelBoot
+			kernelBoot.Container.InitrdPath = ""
+
+			Expect(vmi.Spec.Volumes).ToNot(BeEmpty())
+			Expect(vmi.Spec.Domain.Devices.Disks).ToNot(BeEmpty())
+
+			By("Ensuring VMI can boot")
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
+			Expect(err).ToNot(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(vmi)
+
+			By("Fetching virt-launcher pod")
+			virtLauncherPod := tests.GetRunningPodByVirtualMachineInstance(vmi, vmi.Namespace)
+
+			By("Ensuring VMI is deleted")
+			err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Delete(vmi.Name, &v1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(func() (isVmiDeleted bool) {
```

Comment:
You can use `tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)`

Reply:
Let's keep that for now. The file does not use the normal cleanup procedures. We can switch over the whole file to it at in another PR.

Reply:
I should really start writing `Nit: You can....`